### PR TITLE
User creation & authentication

### PR DESCRIPTION
**Pull request by TiuTalk**
- [x] Create `Accounts` context
- [x] Create `User` model
- [x] Add `User` model validations
- [x] Encrypt user password
- [x] Create `POST /users` request for user creation
- [x] Create `POST /users/authenticate` for user login
- [x] Generate **JWT** for the user

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -22,6 +22,11 @@ config :logger, :console,
   format: "$time $metadata[$level] $message\n",
   metadata: [:user_id]
 
+# Guardian config
+config :dml, Dml.Guardian,
+  issuer: "DML",
+  secret_key: "u49DGdIfepFS21VX6A30S7K32KZOvm54SL/AsvCuSSN1UJM4K6fMR8ImEkmAqcJ5"
```
Comment (mslipper) [inline config/config.exs]: This a production key?

**Comment by mslipper** [comment]
In general this looks fine to me. To confirm my understanding, you're using a third party lib to provide authentication correct?

**Comment by mslipper** [comment]
Just make sure that key gets removed if it is in fact a production key.

File: config/config.exs (hunk 1 above)
Comment (TiuTalk) [reply config/config.exs]: It's a development key, on production it will be loaded from and ENV variable 👍

**Comment by TiuTalk** [comment]
@mslipper yeah, the 3rd party lib ([guardian](https://github.com/ueberauth/guardian)) was used for generating/validating the JWT for the user

**Pull request merged**